Formula/idevicerestore.rb

```diff
@@ -0,0 +1,34 @@
+class Idevicerestore < Formula
+  desc "Command-line application to restore firmware files to iOS devices"
+  homepage "https://libimobiledevice.org/"
+  url "https://github.com/libimobiledevice/idevicerestore/releases/download/1.0.0/idevicerestore-1.0.0.tar.bz2"
+  sha256 "32712e86315397fd2e8999e77a2d2f790c67f6b4aa50d4d1c64cb2c4609836f7"
+  license "LGPL-3.0-only"
+
+  head do
+    url "https://git.libimobiledevice.org/idevicerestore.git", branch: "master"
+    depends_on "autoconf" => :build
+    depends_on "automake" => :build
+    depends_on "libtool" => :build
+  end
+
+  depends_on "pkg-config" => :build
+  depends_on "libimobiledevice"
+  depends_on "libirecovery"
+  depends_on "libplist"
+  depends_on "libzip"
+  depends_on "openssl@1.1"
+
+  def install
+    system "./autogen.sh" if build.head?
+    system "./configure", *std_configure_args,
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
+    system "make", "install"
+  end
+
+  test do
+    system "#{bin}/idevicerestore", "--help"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I have checked how [libimobiledevice](https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/libimobiledevice.rb) and [ideviceinstaller](https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/ideviceinstaller.rb) formula done its tests, just simply running a usage check. For testing its functionality, an iOS device must be connected. As there's not alway an iOS device being connected to the computer for testing if it really able to be functional, printing help is what all we can do.

Reply:
>    If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
>    Same if the software requires a virtual machine, docker instance, etc. to be running.

This also applies for connected devices. Checking existing formula isn't really a good benchmark because we constantly improve our packaging and do not always update older formulae.